### PR TITLE
Report file and log now has a time stamp Vault hardening check enhancements

### PR DESCRIPTION
**Pull request by ediulia**
### Desired Outcome

Main.ps1 now creates report and log files with the time stamp to avoid appending the existing log
Vault hardening now checking the recovery folder content if any file found
bug fixes

### CODE REVIEW

File: CYBRHardeningCheck/Main.ps1 [hunk 1]
```diff
@@ -542,6 +564,17 @@ $outputFile = New-HTMLReportOutput -machineName $machineName -components $(Get-D
 Out-HardeningFolderPath -Path $outputFile -TotalComponentsFound $(Get-DetectedComponents).Name.Count
 
 Write-LogMessage -Type Info -MSG "Hardening Health Check Report located in: $outputFile" -LogFile $LOG_FILE_PATH
-. $outputFile
+# . $outputFile
+
+    if($ReturnReportPath.IsPresent)
+    {
+        # return the report path and skip opening the file interactively
+        EndScript -ReturnData -HardeningReportPath $outputFile
```
Comment (AssafMiron) [inline CYBRHardeningCheck/Main.ps1:572]: @ediulia it seems a bit redundant to use the same switch twice
I don't see why adding it to the EndScript function

File: CYBRHardeningCheck/Main.ps1 [hunk 2]
```diff
@@ -346,9 +363,14 @@ Function New-HTMLReportOutput
 	)
 
 	Begin {
-		$reportDateTime = $(Get-Date -Format $g_DateTimePattern)
+		# Parameter moved to main init
+        # $reportDateTime = $(Get-Date -Format $g_DateTimePattern)
+
 		$htmlFileContent = Get-Content $REPORT_TEMPLATE_PATH
-		$exportFileName = "Hardening_HealthCheck_Report_$($reportDateTime.Replace("/","-").Replace(":","-").Replace(" ","_")).html"
```
Comment (AssafMiron) [inline CYBRHardeningCheck/Main.ps1:351]: Why did you move it from this location?
it is not really needed in the overall script, only here where we create the report file

File: CYBRHardeningCheck/Main.ps1 [hunk 3]
```diff
@@ -187,13 +196,21 @@ Function EndScript
...
 	Write-LogMessage -Type Info -MSG "Script ended" -Footer -LogFile $LOG_FILE_PATH
 
 	# UnLoad loaded modules
-	Remove-ScriptModule $moduleInfos	
+	Remove-ScriptModule $moduleInfos
+
+    # Return the Report Path if ReturnData switch is selected	
+    if($ReturnData.IsPresent)
+    {
+        return $HardeningReportPath
```
Comment (AssafMiron) [inline CYBRHardeningCheck/Main.ps1:212]: You return a value that you received as parameter, why?
If you already had it, use it

File: CYBRHardeningCheck/Main.ps1 [hunk 4]
```diff
@@ -24,17 +25,25 @@ $global:InDebug = $PSBoundParameters.Debug.IsPresent
...
+[string]$global:g_DateTimePattern = "$([System.Globalization.CultureInfo]::CurrentCulture.DateTimeFormat.ShortDatePattern) $([System.Globalization.CultureInfo]::CurrentCulture.DateTimeFormat.LongTimePattern)"
+
+# Set the script time stamp
+$reportDateTime = $(Get-Date -Format $g_DateTimePattern)
+# $reportDateTime = (Get-Date).toString("yyyy-MM-dd_HH_mm_ss")
+
+# Set output files paths
+# $global:LOG_FILE_PATH = "$ScriptLocation\Hardening_HealthCheck.log"
+# Time stamp added to the log name (emilg@segmentech.com)
+$global:LOG_FILE_PATH = "$ScriptLocation\Hardening_HealthCheck_$($reportDateTime.Replace("/","-").Replace(":","-").Replace(" ","_")).log"
```
Comment (AssafMiron) [inline CYBRHardeningCheck/Main.ps1:40]: If the log file and the report needs to use the same file pattern with dates, worth having a parameter to hold that information once for all the places that are using it and concatenating it where it is needed no?

File: CYBRHardeningCheck/Main.ps1 [hunk 5]
```diff
@@ -15,6 +15,7 @@
 [CmdletBinding()]
 param
 (
+    [switch]$ReturnReportPath
```
Comment (AssafMiron) [inline CYBRHardeningCheck/Main.ps1:18]: Maybe the switch should be "DisableAutoReportOpening"?
The logic is either return the path (which we can always do, it is harmless and probably good to add to the log as well
or open the report
so if I narrow it down, it is either opening the report or not
Do you agree?

**Review by AssafMiron — CHANGES_REQUESTED** [review]

**Comment by AssafMiron** [comment]
@ediulia  I resolved some of the conflicts we had on this PR - please be aware

File: CYBRHardeningCheck/Main.ps1 (hunk 2 above)
Comment (ediulia) [reply CYBRHardeningCheck/Main.ps1:351]: @AssafMiron All the Init definitions need to be set before the execution

File: CYBRHardeningCheck/Main.ps1 (hunk 3 above)
Comment (ediulia) [reply CYBRHardeningCheck/Main.ps1:212]: @AssafMiron The parameter is $ReturnData (switch)
And the script will return the Report path instead of opening it.
It will allow to execute the script remotely through WinRm session and retrieve the report location to be able to copy it from PSSession. Example:
#region -------- EXAMPLE: remote execution script block to get the report path --------
$RunRemoteHardeningReport = {
    $ReportPath = S:\servertemp\scripts\Scripts\GLOBAL\CYBRHardeningCheck\Main.ps1 -Verbose -Debug -ReturnReportPath
    return $ReportPath
}

# execute remote scropt block and return the report file path
$RemoteReportPath = Invoke-Command -Session $mysession -ScriptBlock $RunRemoteHardeningReport

# Copy remote report file to the network path
Copy-Item -Path $RemoteReportPath -Destination $NetworkHardeningReportPath -Force -FromSession $mysession -Verbose

File: CYBRHardeningCheck/Main.ps1 (hunk 5 above)
Comment (ediulia) [reply CYBRHardeningCheck/Main.ps1:18]: @AssafMiron if the report filename has the time stamp. It's always good to get it instead of guessing that.
About the switch DisableAutoReportOpening, in general, I have no problem

File: CYBRHardeningCheck/Main.ps1 (hunk 3 above)
Comment (AssafMiron) [reply CYBRHardeningCheck/Main.ps1:212]: I understand the use case and why
But specifically in this function you are not using it like this
The EndScript function is not responsible of opening the file, this happens in the main
So what happens is that you call in main to this EndScript function with a parameter of the HTML report and if the switch is returnData then you return the same HTML report path that I called this script with
but you don't need it, you just need to return that in the main script instead of opening the file (which you did)

**Comment by AssafMiron** [comment]
@ediulia please see #82

**Comment by AssafMiron** [comment]
Closing this PR as we already implemented these changes in a different PR
